### PR TITLE
Add method array_remove.

### PR DESCRIPTION
**Pull request by sfc-gh-lninobrijaldo**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes saved changes [SNOW-1529718)

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

Morgan Stanley: array_remove is available in SQL but there is no corresponding API

### CODE REVIEW

**Comment by sfc-gh-lninobrijaldo** [comment]
I have read the CLA Document and I hereby sign the CLA

File: CHANGELOG.md [hunk 1]
```diff
@@ -64,6 +65,8 @@
 - Fixed a bug in `DataFrame.lineage.trace` to split the quoted feature view's name and version correctly.
 - Fixed a bug in `Column.isin` that caused invalid sql generation when passed an empty list.
 - Fixed a bug that fails to raise NotImplementedError while setting cell with list like item.
+- Fixed a bug in `session.read.csv` that caused an error when setting `PARSE_HEADER = True` in an externally defined file format.
+- Fixed a bug in `pyspark.sql.functions.array_remove` to remove is available in SQL but there is no corresponding API.
```
Comment (sfc-gh-aalam) [inline CHANGELOG.md:69]: move this up to 1.22.0 release section

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,7 +8,8 @@
 
 - Fixed a bug in `session.read.csv` that caused an error when setting `PARSE_HEADER = True` in an externally defined file format.
 - Fixed a bug in query generation from set operations that allowed generation of duplicate queries when children have common subqueries.
-
+- Fixed a bug in `pyspark.sql.functions.array_remove` to remove is available in SQL but there is no corresponding API.
+-
```
Comment (sfc-gh-aalam) [inline CHANGELOG.md:12]: we don't have `pyspark.sql.functions`. Also, this is not a bug. Please remove this log line.

File: CHANGELOG.md (hunk 1 above)
Comment (sfc-gh-aalam) [reply CHANGELOG.md:69]: Again, this is not a bug. Please add it in new feature section

File: src/snowflake/snowpark/functions.py [hunk 3]
```diff
@@ -5333,6 +5333,32 @@ def array_append(array: ColumnOrName, element: ColumnOrName) -> Column:
     return builtin("array_append")(a, e)
 
 
+def array_remove(array: ColumnOrName, element) -> Column:
```
Comment (sfc-gh-aalam) [inline src/snowflake/snowpark/functions.py:5336]: Add type hint for element

File: tests/integ/scala/test_function_suite.py [hunk 4]
```diff
@@ -2823,6 +2824,34 @@ def test_array_append(session):
     )
 
 
+@pytest.mark.skipif(
+    "config.getoption('local_testing_mode', default=False)",
+    reason="array_remove is not yet supported in local testing mode.",
+)
```
Comment (sfc-gh-aalam) [inline tests/integ/scala/test_function_suite.py:2830]: @sfc-gh-aling / @sfc-gh-jrose do we have a jira to support array functions? I believe there are multiple array related functions not supported in local testing mode.

Comment (sfc-gh-jrose) [reply tests/integ/scala/test_function_suite.py:2830]: We do not currently have a jira for this. I can add one to our backlog.

File: CHANGELOG.md (hunk 2 above)
Comment (sfc-gh-lninobrijaldo) [reply CHANGELOG.md]: I added "New Features"  and moved the new the feature to there.

File: CHANGELOG.md (hunk 1 above)
Comment (sfc-gh-lninobrijaldo) [reply CHANGELOG.md:75]: I added the new feature section.

File: src/snowflake/snowpark/functions.py (hunk 3 above)
Comment (sfc-gh-lninobrijaldo) [reply src/snowflake/snowpark/functions.py]: I added type hint for element.

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,10 @@
 
 ### Snowpark Python API Updates
 
+### New Features
+
+- Added support for `functions.array_remove`.
```
Comment (sfc-gh-aalam) [inline CHANGELOG.md:9]:
```suggestion
- Added support for `array_remove` in `snowflake.snowpark.functions`.
```

File: src/snowflake/snowpark/functions.py [hunk 6]
```diff
@@ -5333,6 +5333,32 @@ def array_append(array: ColumnOrName, element: ColumnOrName) -> Column:
     return builtin("array_append")(a, e)
 
 
+def array_remove(array: ColumnOrName, element: ColumnOrName) -> Column:
```
[the remark this answers is not in the document]
Comment (sfc-gh-lninobrijaldo) [reply src/snowflake/snowpark/functions.py]: I did the change.

File: CHANGELOG.md (hunk 5 above)
Comment (sfc-gh-lninobrijaldo) [reply CHANGELOG.md]: I did it.

**Review by sfc-gh-aalam — APPROVED** [review]

**Review by sfc-gh-yuwang — APPROVED** [review]

**Pull request merged**